frontend/src/pages/cert-manager/PkiSyncDetailsByIDPage/components/PkiSyncActionTriggers.tsx

```diff
@@ -0,0 +1,314 @@
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  IconButton,
+  Tooltip
+} from "@app/components/v2";
+import { ROUTE_PATHS } from "@app/const/routes";
+import { ProjectPermissionSub } from "@app/context";
+import { ProjectPermissionPkiSyncActions } from "@app/context/ProjectPermissionContext/types";
+import { PKI_SYNC_MAP } from "@app/helpers/pkiSyncs";
+import { usePopUp, useToggle } from "@app/hooks";
+import {
+  TPkiSync,
+  useTriggerPkiSyncSyncCertificates,
+  useUpdatePkiSync
+} from "@app/hooks/api/pkiSyncs";
+import { IntegrationsListPageTabs } from "@app/types/integrations";
+
+type Props = {
+  pkiSync: TPkiSync;
+};
+
+export const PkiSyncActionTriggers = ({ pkiSync }: Props) => {
+  const { destination, subscriberId, projectId, id } = pkiSync;
+
+  const navigate = useNavigate();
+  const { popUp, handlePopUpOpen, handlePopUpToggle } = usePopUp([
+    "importCertificates",
+    "removeCertificates",
+    "deleteSync"
+  ] as const);
+
+  const [isIdCopied, setIsIdCopied] = useToggle(false);
+
+  // API mutations
+  const triggerSyncMutation = useTriggerPkiSyncSyncCertificates();
+  const updatePkiSyncMutation = useUpdatePkiSync();
+
+  const destinationName = PKI_SYNC_MAP[destination].name;
+
+  const handleCopyId = useCallback(() => {
+    setIsIdCopied.on();
+    navigator.clipboard.writeText(pkiSync.id);
+
+    createNotification({
+      text: "PKI Sync ID copied to clipboard",
+      type: "info"
+    });
+
+    const timer = setTimeout(() => setIsIdCopied.off(), 2000);
+    return () => clearTimeout(timer);
+  }, [pkiSync.id, setIsIdCopied]);
+
+  const handleTriggerSync = useCallback(async () => {
+    try {
+      await triggerSyncMutation.mutateAsync({
+        syncId: id,
+        projectId,
+        destination
+      });
+      createNotification({
+        text: "PKI sync job queued successfully",
+        type: "success"
+      });
+    } catch (error) {
+      console.error("Failed to trigger sync:", error);
+      createNotification({
+        text: "Failed to trigger PKI sync",
+        type: "error"
+      });
+    }
+  }, [triggerSyncMutation, id, projectId]);
+
+  const handleToggleAutoSync = useCallback(async () => {
+    try {
+      await updatePkiSyncMutation.mutateAsync({
+        syncId: id,
+        projectId,
+        destination,
+        isAutoSyncEnabled: !pkiSync.isAutoSyncEnabled
+      });
+      createNotification({
+        text: `Auto-sync ${pkiSync.isAutoSyncEnabled ? "disabled" : "enabled"} successfully`,
+        type: "success"
+      });
+    } catch (error) {
+      console.error("Failed to toggle auto-sync:", error);
+      createNotification({
+        text: "Failed to toggle auto-sync",
+        type: "error"
+      });
+    }
+  }, [updatePkiSyncMutation, id, projectId, pkiSync.isAutoSyncEnabled]);
+
+  const permissionSubject = subscriberId
+    ? subject(ProjectPermissionSub.PkiSyncs, { subscriberId })
+    : ProjectPermissionSub.PkiSyncs;
```

Comment:
I noticed that we don't do it like this for the other files... which one is correct? For the other ones it looks like we just default the subscriberId to empty string